### ISSUE TITLE
ToggleGroupControl: isBlock + isAdaptiveWidth not working with icons

### ISSUE THREAD

**Issue by xconsau** [WordPress/gutenberg#45988]
### Description

When using the `ToggleGroupControlOptionIcon`, the `isBlock` and `isAdaptiveWidth` combination on the `ToggleGroupControl` doesn't work. The expected appearance would be a full width parent container with equal width distributed inner items. Whereas it shows as inline items with auto width.

The two props work well when used with the text items with `ToggleGroupControlOption`, but not with the `ToggleGroupControlOptionIcon`.

_I understand that this feature is still experimental and is subject to drastic changes. So please disregard the issue if this change was deliberate._

### Step-by-step reproduction instructions

In a test Gutenberg block, try adding the following component code in edit.js file:

```jsx
<ToggleGroupControl
    label={__( 'Title Align' )}
    value={attributes.title_align}
    onChange = {(newval) => setAttributes({title_align: newval })}
    isBlock
    isAdaptiveWidth
>
    <ToggleGroupControlOptionIcon
        value="left"
        icon={ alignLeft }
        showTooltip={ true }
        aria-label={__( 'Left' )}
    />
    <ToggleGroupControlOptionIcon
        value="center"
        icon={ alignCenter }
        showTooltip={ true }
        aria-label={__( 'Center' )}
    />
    <ToggleGroupControlOptionIcon
        value="right"
        icon={ alignRight }
        showTooltip={ true }
        aria-label={__( 'Right' )}
    />
    <ToggleGroupControlOptionIcon
        value="justify"
        icon={ alignJustify }
        showTooltip={ true }
        aria-label={__( 'Justify' )}
    />
</ToggleGroupControl>
```

### Screenshots, screen recording, code snippet

Below are the screen captures with diferent settings:

1. With `isBlock` and `isAdaptiveWidth` enabled:

![toggleGroupControl3](https://user-images.githubusercontent.com/11784237/203489320-0580facf-6e4f-4cf6-828d-b3daa043e75c.gif)

2. Without `isBlock`:

![toggleGroupControl2](https://user-images.githubusercontent.com/11784237/203487518-f2e2eb07-022e-49a1-842b-a8a65eacbb2c.gif)

3. Expected appearance:

![toggleGroupControl1](https://user-images.githubusercontent.com/11784237/203487601-c7e0a4b3-498c-4964-afdc-cb17316511e7.gif)

### Environment info

- WordPress 6.1.1 and Gutenberg as shipped with this version.
- Chrome version 107.0.5304.107
- Desktop with Windows 10

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by mirka** [comment]
We've recently been reworking this component to support newer design requirements, and it's true that the API is not fully coherent at the moment.

According to the style guidelines, a ToggleGroupControl with icon options should never be `isBlock`. On the other hand, a ToggleGroupControl with text options should almost always be `isBlock`. 

`isAdaptiveWidth` is another questionable prop. Contrary to what you've written, `isAdaptiveWidth=true` is supposed to make the items _not_ equal width — but yes, the prop description kind of implies the inverse! Therefore, it only takes effect when `isBlock=false` (i.e. it shouldn't be used in combination with `isBlock` to begin with). The `isAdaptiveWidth` prop isn't used anywhere in the Gutenberg app, so it's likely that this prop was never backed by an actual app design requirement. It might even be good to remove the prop altogether.

## tl;dr

- Gutenberg style guidelines recommend that you **not** use `isBlock`/`isAdaptiveWidth` with icon options.
- We're in the process of ironing out incoherent prop combinations like this, so people don't have to know what the guidelines are.

**Comment by mirka** [comment]
## Action items

- [ ] Consider retiring the `isAdaptiveWidth` prop
- [ ] Consider splitting ToggleGroupControl into two components — one for text options and one for icon options

**Comment by xconsau** [comment]
Thank you @mirka for the clarification. Much appreciated.
